core/src/main/java/com/datastax/oss/driver/internal/core/cql/PagingIterableSpliterator.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.cql;
+
+import com.datastax.oss.driver.api.core.PagingIterable;
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import net.jcip.annotations.NotThreadSafe;
+
+/**
+ * A Spliterator for {@link PagingIterable} instances that splits the stream in chunks of equal
+ * size.
+ *
+ * @param <ElementT> The element type of the underlying stream.
+ */
+@NotThreadSafe
+public class PagingIterableSpliterator<ElementT> implements Spliterator<ElementT> {
+
+  /**
+   * Creates a new {@link Builder} for {@link PagingIterableSpliterator} instances.
+   *
+   * @param iterable The {@link PagingIterable} to create a spliterator for.
+   * @param <ElementT> The iterable's element type
+   * @return a new {@link Builder}.
+   */
+  @NonNull
+  public static <ElementT> Builder<ElementT> builder(@NonNull PagingIterable<ElementT> iterable) {
```

Comment:
Nit: do we need that level of detail in javadocs for internal classes? This block doesn't really tell anything that isn't obvious from the method signature.